sklearn/model_selection/_prediction.py

```diff
@@ -0,0 +1,339 @@
+from inspect import signature
+
+import numpy as np
+
+from ..base import clone
+from ..base import BaseEstimator
+from ..base import ClassifierMixin
+from ..base import MetaEstimatorMixin
+from ..exceptions import NotFittedError
+from ..metrics import balanced_accuracy_score
+from ..metrics import confusion_matrix
+from ..metrics import roc_curve
+from ..preprocessing import LabelEncoder
+from ..utils import check_array
+from ..utils.multiclass import check_classification_targets
+from ..utils.multiclass import type_of_target
+from ..utils.validation import check_is_fitted
+
+
+class CutoffClassifier(MetaEstimatorMixin, ClassifierMixin, BaseEstimator):
+    """Decision threshold calibration for binary classification.
+
+    Estimator that calibrates the decision threshold (cutoff point) that is
+    used for prediction. The methods for picking cutoff points make use of
+    traditional binary classification evaluation statistics such as the true
+    positive and true negative rates or any metrics accepting true labels and
+    the output of a scoring function from a scikit-learn estimator.
+
+    Parameters
+    ----------
+    base_estimator : estimator object
+        The classifier, fitted or not fitted, from which we want to optimize
+        the decision threshold used during `predict`.
+
+    objective_metric : {"tpr", "tnr"}, ndarray of shape (2, 2) or callable, \
```

Comment:
If should also include `"f1"` and `"balanced_accuracy"` and maybe others such as `"fbeta"` if we also include a new dedicted "beta" constructor param.

Reply:
I was thinking to let the user pass a skelarn.metric callable. It makes it easy because you will never need a `objective_value` for this case and only need an `objective_value` when the a `str` is provided because we will work with predefined search.

Reply:
I would still make it possible to pass strings for metrics that can be efficiently computed from the output of _binary_clf_curve. This private tool could actually be extended to `_binary_clf_confusion_matrix_curve` that would compute the CM for each threshold.

Reply:
> "fbeta" if we also include a new dedicated "beta" constructor param

This could be pass to `objective_metric_param` indeed.

Reply:
> I would still make it possible to pass strings for metrics that can be efficiently computed from the output of _binary_clf_curve. This private tool could actually be extended to _binary_clf_confusion_matrix_curve that would compute the CM for each threshold.

OK. I will first make the documentation with the naive approach and then, I will improve on the efficiency.